### ISSUE TITLE
Memory usage of GNNs

### ISSUE THREAD

**Issue by legohyl** [pyg-team/pytorch_geometric#1415]
## ❓ Questions & Help

hey @rusty1s , is there a way to roughly calculate the memory usage of GNN modules? It looks like if I have a graph that's about 900 nodes large, not fully connected, using 3 iterations of NNConv, consumes about 3.4Gb or so during training. Is there a way to check which parts are consuming a large amount of memory and how to reduce them? This is with 64-dim node representations.

**Comment by rusty1s** [comment]
I think you can make use of underlying PyTorch function to achieve that, see [here](https://pytorch.org/docs/stable/cuda.html#memory-management). However, `NNConv` is known to be very memory-inefficient (and as far as I know, there is no way around it), since it computes an individual weight matrix for each edge.

**Comment by legohyl** [comment]
Ah I see okay. It appears like the hooks generated for the backward pass are causing a large spike in memory.

**Comment by balakkvj** [comment]
Along very similar lines, my graphs are huge, each over 10,000 nodes and 100,000 edges; with an embedding size of 1200, memory has been a big impediment. In fact we decided to cut our dimension to 64 as a result. Apart from the cuda memory you just pointed out, and the batch size, are there any hidden gems in torch_geometric I could use?

Thanks so much.

**Comment by rusty1s** [comment]
Please have a look [here](https://pytorch-geometric.readthedocs.io/en/latest/notes/sparse_tensor.html).
